x-pack/elastic-agent/docs/elastic-agent-command-line.asciidoc

```diff
@@ -8,40 +10,276 @@
 
 beta[]
 
-The `elastic-agent run` command provides flags that alter the behavior of an
-agent:
+{agent} provides commands for running the agent and doing common tasks:
 
-`--path.home`::
-The home directory of the {agent}. `path.home` determines the location of the
-configuration files and data directory.
+* <<elastic-agent-enroll-command,enroll>>
+* <<elastic-agent-help-command,help>>
+* <<elastic-agent-inspect-command,inspect>>
+* <<elastic-agent-run-command,run>>
+* <<elastic-agent-version-command,version>>
+
+++++
+<hr>
+++++
+
+[discrete]
+[[elastic-agent-enroll-command]]
+== elastic-agent enroll
+
+Enroll the {elastic-agent} in {fleet}.
+
+We recommend that you run this command as the root user because some
+integrations require root privileges to collect sensitive data. This command
+overwrites the `elastic-agent.yml` file in the agent directory.
+
+[discrete]
+=== Synopsis
+
+[source,shell]
+----
+elastic-agent enroll <kibana_url> <enrollment_token> [--ca-sha256 <string>]
+                     [--certificate-authorities <string>] [--force] [--help]
+                     [--insecure][global-flags]
+----
+
+[discrete]
+=== Options
+
+`kibana_url`::
+Required. URL of the {kib} endpoint where {fleet} is running.
+
+`enrollment_token`::
+Required. Enrollment token generated by {fleet}. You can use the same
+enrollment token for multiple agents.
+
+`--ca-sha256 <string>`::
+Comma-separated list of certificate authority hash pins used for certificate
+verification.
+
+`--certificate-authorities <string>`::
+Comma-separated list of root certificates used for server verification.
+
+`--force`::
+Force overwrite of current configuration without prompting for confirmation.
+This flag is helpful when using automation software or scripted deployments.
+
+`--help`::
+Show help for the `enroll` command.
+
+`--insecure`::
+Allow an insecure connection to {kib}. We strongly recommend that you use a
```

Comment:
We should probably add a bit more details in the future what insuecure means, like API Keys are sent in clear text.